### PR TITLE
Trianglesets

### PR DESCRIPTION
**Pull request by gangatp**
Rebased the old branch with new develop. Targeting for the next release. 

This is the release of the core spec
https://github.com/3MFConsortium/spec_core/releases/tag/1.3.0

- ToDo
- Add unit tests
- Add examples

### CODE REVIEW

File: Source/Model/Classes/NMR_ModelTriangleSet.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	{
+		m_sTriangles.erase(nTriangleIndex);
+	}
+
+	PModelTriangleSet CModelTriangleSet::duplicate(const std::string& sIdentifier)
+	{
+		auto pTriangleSet = std::make_shared<CModelTriangleSet>(m_sName, sIdentifier);
+
+		for (auto nTriangleIndex : m_sTriangles)
+			pTriangleSet->m_sTriangles.insert(nTriangleIndex);
```
Comment (3dJan) [inline Source/Model/Classes/NMR_ModelTriangleSet.cpp:91]:
```suggestion
		{
			pTriangleSet->m_sTriangles.insert(nTriangleIndex);
		}
```

File: Source/Model/Classes/NMR_ModelTriangleSet.cpp [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		return pTriangleSet;
+	}
+
+	void CModelTriangleSet::merge(PModelTriangleSet pTriangleSet)
+	{
+		if (pTriangleSet.get() == nullptr)
+			throw CNMRException(NMR_ERROR_INVALIDPARAM);
+
+		for (auto nTriangleIndex : pTriangleSet->m_sTriangles)
+			m_sTriangles.insert(nTriangleIndex);
```
Comment (3dJan) [inline Source/Model/Classes/NMR_ModelTriangleSet.cpp:102]:
```suggestion
		{
			m_sTriangles.insert(nTriangleIndex);
		}
```

**Review by 3dJan — COMMENTED** [review]
There are not tests yet, that makes it hard to check if the implementation matches the spec.

**Comment by gangatp** [comment]
> There are not tests yet, that makes it hard to check if the implementation matches the spec.

Hi Jan, thanks for reviewing. I will add unit test and examples to this PR.

**Comment by gangatp** [comment]
Hi Team, I added a basic unit test to check the functionality.

**Comment by gangatp** [comment]
I will do a follow up PR to fix the code coverage.

**Pull request merged**